singer-io/tap-quickbooks
# [TDL-15661] Implement dev mode for 2.latest

# Description of change
Add support for -D / --dev cli param,

- Prevent tokens expiry while running in dev-mode
- Separate function to create the `session` depending on dev-mode enabled/disabled.
    -  Add two more new fields `access_token` and `expires_at` to the config whenever the tokens are refreshed in normal flow.
- Unit tests to verify the flow of dev-mode.

# Manual QA steps
- First run the tap normally using the standard params
- Check the config file for added keys `access_token` & `expires_in`
- Run the tap using the '-D' or '--dev' param
- Check the log life for dev-mode related logs

 
# Risks
 - Tap will exit for existing connections when run in dev mode if `access_token` and `expires_at` is not available/valid.
 
# Rollback steps
 - revert this branch
